File: 2a/cpp/funkcje.cpp
/*
 * szkielet.cpp
 *
 */


#include <iostream>
/* io input/output czyli biblioteka wejścia/wyjścia w pythonie nie trzeba tego importować
 * zawsze musi być main bo to nazwa funkcji*/

using namespace std;

void suma(int a, int b) {
    int wynik = a + b;
    cout << wynik;
}

int main(int argc, char **argv)
{
    int a, b;
    a = b = 0;
   
    cout << "Podaj liczbę 1: ";
    cin >> a;
    cout << a;

    cout << endl << "Podaj liczbę 2: ";
    cin >> b;
    ,
    cout << b;
   
    cout << endl << "Suma: " << a + b << endl;
    cout << endl << "Różnica: " << a - b << endl;
    cout << endl << "Iloczyn: " << a * b << endl;
    cout << endl << "Iloraz: " << a / b << endl;
    return 0;
}

//DRY dont repeat yourself

File: 2a/python/funkcje.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#duck typing
#wszystko co pobierane jest z domyślnego wejścia z terminala jest znakiem
# zasięŋ zmiennych: zasięg lokalny (ograniczony np. funkcją)

def suma(a, b):
    return a + b
    
    
def różnica(a, b):
    return a - b
    
    
def iloczyn(a, b):
    return a * b    


def iloraz(a, b):
    return a / b
    
    
def main(args):
    a = int(input("Podaj liczbę 1: "))
    b = int(input("Podaj liczbę 2: "))
    print(a)
    print(b)
    
    print("Suma: ", suma(a, b))
    print("Różnica: ", różnica(a, b))
    print("Iloczyn: ", iloczyn(a, b))
    print("Iloraz: ", iloraz(a, b))

    
    return 0    

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
